pennylane/ops/qubit/matrix_ops.py

```diff
@@ -264,18 +270,30 @@ def compute_decomposition(U: TensorLike, wires: WiresLike):
 
         return super(QubitUnitary, QubitUnitary).compute_decomposition(U, wires=wires)
 
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_sparse_matrix(self) -> bool:
+        return self._issparse
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self) -> bool:
+        return not self._issparse
+
     # pylint: disable=arguments-renamed, invalid-overridden-method
     @property
     def has_decomposition(self) -> bool:
         return len(self.wires) < 3
 
     def adjoint(self) -> "QubitUnitary":
-        U = self.matrix()
-        if isinstance(U, csr_matrix):
+        if self.has_matrix:
+            U = self.matrix()
+            return QubitUnitary(qml.math.moveaxis(qml.math.conj(U), -2, -1), wires=self.wires)
+        else:
+            U = self.sparse_matrix()
             adjoint_sp_mat = U.conjugate().transpose()
             # Note: it is necessary to explicitly cast back to csr, or it will be come csc
             return QubitUnitary(csr_matrix(adjoint_sp_mat), wires=self.wires)
```

Comment:
```suggestion
        U = self.sparse_matrix()
        adjoint_sp_mat = U.conjugate().transpose()
        # Note: it is necessary to explicitly cast back to csr, or it will be come csc
        return QubitUnitary(csr_matrix(adjoint_sp_mat), wires=self.wires)
```

Reply:
no need for `else` after a return.